src/lib/dev-environment/dev-environment-core.js

```diff
@@ -504,20 +488,16 @@ async function updateWordPressImage( instancePath, options ) {
 
 	// If the user takes the new WP version path
 	if ( confirm.upgrade ) {
-		console.log( 'Upgrading from: ' + chalk.yellow( match.ref ) + ' to:' );
+		console.log( 'Upgrading from: ' + chalk.yellow( current.wordpress.tag ) + ' to:' );
 
 		// Select a new image
 		const choice = await promptForComponent( 'wordpress' );
 		const version = filteredVersions.find( ( { tag } ) => tag.trim() === choice.tag.trim() );
-		const selectedImage = `ghcr.io/automattic/vip-container-images/wordpress:${ version.tag }`;
-
-		// Change the lando file and rebuild.
-		const data = fs.readFileSync( landoFile, { encoding: 'utf8', flag: 'r' } );
-		const edit = data.replace( /ghcr\.io\/.*wordpress\:.*\d+\.\d+(?:\.\d+)?/g, selectedImage );
-		fs.writeFileSync( landoFile, edit );
 
-		// Stage for rebuild
+		// // Write new data and stage for rebuild
+		current.wordpress.tag = version.tag;
 		options.skipRebuild = false;
```

Comment:
Hmm this mutating of argument can be a bit confusing. What do you think of returning boolean from `updateWordPressImage`. True if it was changed.

Then in the caller you can do:
```
const updated = await updateWordPressImage( slug, options );

	if ( options.skipRebuild && !updated) {
		await landoStart( instancePath );
	} else {
		await landoRebuild( instancePath );
	}
```